src/src/rx_main.cpp

```diff
@@ -326,24 +330,23 @@ bool ICACHE_RAM_ATTR HandleSendTelemetryResponse()
             NextTelemetryType = ELRS_TELEMETRY_TYPE_LINK;
         }
 
-        TelemetrySender.GetCurrentPayload(&packageIndex, &maxLength, &data);
-        Radio.TXdataBuffer[1] = (packageIndex << ELRS_TELEMETRY_SHIFT) + ELRS_TELEMETRY_TYPE_DATA;
-        Radio.TXdataBuffer[2] = maxLength > 0 ? *data : 0;
-        Radio.TXdataBuffer[3] = maxLength >= 1 ? *(data + 1) : 0;
-        Radio.TXdataBuffer[4] = maxLength >= 2 ? *(data + 2) : 0;
-        Radio.TXdataBuffer[5] = maxLength >= 3 ? *(data + 3): 0;
-        Radio.TXdataBuffer[6] = maxLength >= 4 ? *(data + 4): 0;
+        TelemetrySender.GetCurrentPayload(&iter, &maxLength, &data);
+        otaPktPtr->tlm_dl.packageIndex = iter;
+        otaPktPtr->tlm_dl.type = ELRS_TELEMETRY_TYPE_DATA;
+        for (iter = 0; iter < sizeof(otaPktPtr->tlm_dl.payload) && iter < maxLength; iter++)
+            otaPktPtr->tlm_dl.payload[iter] = *data++;
     }
 
-    uint16_t crc = ota_crc.calc(Radio.TXdataBuffer, 7, CRCInitializer);
-    Radio.TXdataBuffer[0] |= (crc >> 6) & 0b11111100;
-    Radio.TXdataBuffer[7] = crc & 0xFF;
+    uint16_t crc = ota_crc.calc(
+        (uint8_t*)otaPktPtr, OTA_CRC_CALC_LEN, CRCInitializer);
+    otaPktPtr->crcHigh = (crc >> 8);
+    otaPktPtr->crcLow  = crc;
 
 #if defined(Regulatory_Domain_EU_CE_2400)
     if (ChannelIsClear())
 #endif
     {
-        Radio.TXnb();
+        Radio.TXnb((uint8_t*)otaPktPtr, sizeof(*otaPktPtr));
```

Comment:
Payload size cleanup will be my next PR. That struct size is checked to be the same 8B atm.
I will remove all that and will use the struct size only to keep the maintaining simplier... I can include thay into this PR but I think it would be cleaner using separate PRs